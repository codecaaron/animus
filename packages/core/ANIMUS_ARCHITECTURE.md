# Animus.ts Architecture Explanation

## Core Concept

Animus is a **type-state machine** that guides you through building styled components step-by-step. It uses TypeScript's type system to enforce that you configure your components in a specific order, making it impossible to make mistakes.

## The Unique Architecture

The most interesting aspect is the **backwards inheritance chain**:

```
Animus → AnimusWithBase → AnimusWithVariants → AnimusWithStates → AnimusWithSystem → AnimusWithAll
```

Each class extends the one *after* it in the chain, which seems backwards but is actually brilliant. Here's why:

- `Animus` (the starting point) inherits from `AnimusWithBase`
- `AnimusWithBase` inherits from `AnimusWithVariants`
- And so on...

This means the first class has access to all the methods defined in later classes, but those methods only become *available* to use when you reach the right stage.

## How It Works - Step by Step

### 1. Starting Point: `Animus` (lines 331-349)
```typescript
const builder = new Animus(propRegistry, groupRegistry);
```
- Takes your prop definitions and group configurations
- Only exposes the `.styles()` method
- Creates a parser for your props automatically

### 2. After calling `.styles()`: `AnimusWithBase` (lines 287-329)
```typescript
builder.styles({ padding: '20px', color: 'blue' })
```
- Now you can call `.variant()` to add design variations
- The base styles are locked in

### 3. After calling `.variant()`: `AnimusWithVariants` (lines 226-285)
```typescript
.variant({
  prop: 'size',
  variants: {
    small: { fontSize: '14px' },
    large: { fontSize: '20px' }
  }
})
```
- Can add multiple variants
- Now `.states()` method becomes available

### 4. After calling `.states()`: `AnimusWithStates` (lines 184-224)
```typescript
.states({
  disabled: { opacity: 0.5, cursor: 'not-allowed' },
  loading: { cursor: 'wait' }
})
```
- Defines boolean states (props that are true/false)
- Unlocks `.groups()` method

### 5. After calling `.groups()`: `AnimusWithSystem` (lines 140-182)
```typescript
.groups({ spacing: true, colors: true })
```
- Enables predefined groups of props
- Now you can add custom props with `.props()`

### 6. Final Stage: `AnimusWithAll` (lines 24-138)
This is where the magic happens. You can now:

- `.asElement('button')` - Create a styled HTML element
- `.asComponent(MyComponent)` - Style an existing React component  
- `.build()` - Get the raw styling function
- `.extend()` - Create variations of your styled component

## Key Features

### Type Safety Throughout
The 8 generic parameters track everything:
- `PropRegistry` - All your prop definitions
- `GroupRegistry` - Prop groupings
- `BaseParser` - How props convert to CSS
- `BaseStyles` - Your base CSS
- `Variants` - Design variations
- `States` - Boolean state styles
- `ActiveGroups` - Which groups are enabled
- `CustomProps` - Additional custom props

### Smart Prop Forwarding (lines 76-87)
```typescript
shouldForwardProp: (prop) => 
  isPropValid(prop) && !propNames.includes(prop)
```
- System props (like `spacing`, `colors`) are consumed and turned into styles
- Valid HTML attributes are passed through
- Invalid props are filtered out

### The Build Process (lines 99-137)
When you call `.build()`, it:
1. Merges your parser config with custom props
2. Creates a stylist that combines all your configurations
3. Returns a function that generates CSS objects
4. Handles variants, states, and system props automatically

### Extension System
The `.extend()` method creates an `AnimusExtended` instance that can break the rules - you can modify any part of the configuration in any order. This provides an escape hatch when needed.

## Why This Design?

1. **Impossible to Misuse** - You can't call methods in the wrong order
2. **Perfect IntelliSense** - Your IDE knows exactly what methods are available at each step
3. **Type Safe** - All configurations are fully typed
4. **Flexible When Needed** - The extend system provides flexibility
5. **Clean API** - The forced order mirrors how we think about styling (base → variants → states → props)

## Example Usage

```typescript
const Button = animus
  .styles({ 
    padding: '10px 20px',
    borderRadius: '4px' 
  })
  .variant({
    prop: 'variant',
    variants: {
      primary: { background: 'blue', color: 'white' },
      secondary: { background: 'gray', color: 'black' }
    }
  })
  .states({
    disabled: { opacity: 0.6 }
  })
  .asElement('button');

// Usage:
<Button variant="primary" disabled>Click me</Button>
```

## The Backwards Inheritance Explained

The backwards inheritance pattern is the key innovation. Instead of:
- Base class with few methods → Derived class with more methods (traditional OOP)

Animus does:
- Starting class that inherits everything → But methods are selectively exposed

This creates a "progressive revelation" pattern where:
1. You start with minimal options (just `.styles()`)
2. Each method call returns a new class instance
3. That new instance exposes the next logical methods
4. Invalid sequences are impossible at the type level

## Integration with Emotion

Animus builds on top of Emotion (a CSS-in-JS library):
- Uses `@emotion/styled` to create the actual styled components
- Uses `@emotion/is-prop-valid` to filter props intelligently
- The final output is a standard Emotion styled component
- Full compatibility with themes, SSR, and other Emotion features

## Performance Considerations

- All type checking happens at compile time - zero runtime overhead
- The inheritance chain is resolved during build
- Final components are standard Emotion components
- Parser and stylist are created once and reused

## Summary

This architecture ensures that every styled component is configured correctly, with all the power of CSS-in-JS but none of the footguns! The type-state machine pattern combined with backwards inheritance creates a unique developer experience where the API guides you to success.