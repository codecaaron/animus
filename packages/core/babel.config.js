module.exports = {
  extends: "../../babel.config.js",
  presets: ["@babel/preset-typescript"],
  include: ["./src/**/*.ts", "./src/**/*.tsx"],
  ignore: ["./**/*.d.ts", "__tests__"],
};
