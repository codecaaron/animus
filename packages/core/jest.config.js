module.exports = require('../../jest.config.base')('core', 'jsdom');
