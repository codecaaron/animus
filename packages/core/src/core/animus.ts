import { merge } from 'lodash';

import { Parser, Prop, SystemProps, TransformerMap } from '../types/config';
import {
  AbstractProps,
  CSSPropMap,
  CSSProps,
  CSSObject,
  ThemeProps,
} from '../types/props';

import { Arg, PropConfig } from '../types/config';
import { createStylist } from '../styles/createStylist';
import { create } from './create';
import styled from '@emotion/styled';

export class AnimusWithAll<
  C extends PropConfig,
  P extends Parser<TransformerMap<C['props']>>,
  Base extends CSSProps<AbstractProps, SystemProps<P>>,
  Variants extends Record<
    string,
    {
      prop?: any;
      defaultVariant?: any;
      base?: CSSProps<AbstractProps, SystemProps<P>>;
      variants: CSSPropMap<AbstractProps, SystemProps<P>>;
    }
  >,
  States extends CSSPropMap<AbstractProps, SystemProps<P>>,
  G extends (keyof C['groups'])[] | never[]
> {
  props = {} as C;
  parser = {} as P;
  groups = [] as G;
  base = {} as Base;
  statesConfig = {} as States;
  variants = {} as Variants;

  constructor(
    props: C,
    parser: P,
    base: Base,
    variants: Variants,
    states: States,
    groups: G
  ) {
    this.props = props;
    this.parser = parser;
    this.groups = groups;
    this.base = base;
    this.statesConfig = states;
    this.variants = variants;
  }

  asComponent<T extends keyof JSX.IntrinsicElements>(component: T) {
    type Props = ThemeProps<
      {
        [K in keyof Arg<P>]?: Arg<P>[K];
      } & { [K in keyof Variants]?: keyof Variants[K]['variants'] } & {
        [K in keyof States]?: boolean;
      }
    >;

    const handler = createStylist(
      this.parser,
      this.base,
      this.variants,
      this.statesConfig,
      {}
    ) as (props: { [K in keyof Props]: Props[K] }) => CSSObject;

    return styled(component)(handler);
  }

  build() {
    type Props = ThemeProps<
      {
        [K in keyof Arg<P>]?: Arg<P>[K];
      } & { [K in keyof Variants]?: keyof Variants[K]['variants'] } & {
        [K in keyof States]?: boolean;
      }
    >;

    const handler = createStylist(
      this.parser,
      this.base,
      this.variants,
      this.statesConfig,
      {}
    ) as (props: Props) => CSSObject;

    return handler;
  }
}

class AnimusWithSystem<
  C extends PropConfig,
  P extends Parser<TransformerMap<C['props']>>,
  Base extends CSSProps<AbstractProps, SystemProps<P>>,
  Variants extends Record<
    string,
    {
      prop?: any;
      defaultVariant?: any;
      base?: CSSProps<AbstractProps, SystemProps<P>>;
      variants: CSSPropMap<AbstractProps, SystemProps<P>>;
    }
  >,
  States extends CSSPropMap<AbstractProps, SystemProps<P>>,
  G extends (keyof C['groups'])[] | never[]
> extends AnimusWithAll<C, P, Base, Variants, States, G> {
  constructor(
    props: C,
    parser: P,
    base: Base,
    variants: Variants,
    states: States,
    groups: G
  ) {
    super(props, parser, base, variants, states, groups);
  }

  customProps<ExtraProps extends Record<string, Prop>>(config: ExtraProps) {
    const newProps = { ...this.props.props, ...config } as C['props'] &
      ExtraProps;
    const parser = create(newProps);
    return new AnimusWithAll(
      { ...this.props, props: newProps },
      parser,
      this.base,
      this.variants,
      this.statesConfig,
      this.groups
    );
  }
}

class AnimusWithStates<
  C extends PropConfig,
  P extends Parser<TransformerMap<C['props']>>,
  Base extends CSSProps<AbstractProps, SystemProps<P>>,
  Variants extends Record<
    string,
    {
      prop?: any;
      defaultVariant?: any;
      base?: CSSProps<AbstractProps, SystemProps<P>>;
      variants: CSSPropMap<AbstractProps, SystemProps<P>>;
    }
  >,
  States extends CSSPropMap<AbstractProps, SystemProps<P>>
> extends AnimusWithSystem<C, P, Base, Variants, States, []> {
  constructor(
    props: C,
    parser: P,
    base: Base,
    variants: Variants,
    states: States
  ) {
    super(props, parser, base, variants, states, []);
  }

  systemProps<PickedGroups extends keyof C['groups']>(
    config: Record<PickedGroups, true>
  ) {
    return new AnimusWithSystem(
      this.props,
      this.parser,
      this.base,
      this.variants,
      this.statesConfig,
      Object.keys(config) as PickedGroups[]
    );
  }
}

class AnimusWithVariants<
  C extends PropConfig,
  P extends Parser<TransformerMap<C['props']>>,
  Base extends CSSProps<AbstractProps, SystemProps<P>>,
  Variants extends Record<
    string,
    {
      prop?: any;
      defaultVariant?: any;
      base?: CSSProps<AbstractProps, SystemProps<P>>;
      variants: CSSPropMap<AbstractProps, SystemProps<P>>;
    }
  >
> extends AnimusWithStates<C, P, Base, Variants, {}> {
  constructor(props: C, parser: P, base: Base, variants: Variants) {
    super(props, parser, base, variants, {});
  }

  states<Props extends AbstractProps>(
    config: CSSPropMap<Props, SystemProps<P>>
  ) {
    return new AnimusWithStates(
      this.props,
      this.parser,
      this.base,
      this.variants,
      config
    );
  }

  variant<
    Keys extends keyof Props,
    Base extends AbstractProps,
    Props extends Record<Keys, AbstractProps>,
    PropKey extends Readonly<string> = 'variant'
  >(options: {
    prop?: PropKey;
    defaultVariant?: keyof Props;
    base?: CSSProps<Base, SystemProps<P>>;
    variants: CSSPropMap<Props, SystemProps<P>>;
  }) {
    type NextVariants = Variants & Record<PropKey, typeof options>;
    const prop = options.prop || 'variant';

    return new AnimusWithVariants(
      this.props,
      this.parser,
      this.base,
      merge(this.variants, { [prop]: options }) as NextVariants
    );
  }
}

class AnimusWithBase<
  C extends PropConfig,
  P extends Parser<TransformerMap<C['props']>>,
  Base extends CSSProps<AbstractProps, SystemProps<P>>
> extends AnimusWithVariants<C, P, Base, {}> {
  constructor(props: C, parser: P, base: Base) {
    super(props, parser, base, {});
  }
  variant<
    Keys extends keyof Props,
    Base extends AbstractProps,
    Props extends Record<Keys, AbstractProps>,
    PropKey extends Readonly<string> = 'variant'
  >(options: {
    prop?: PropKey;
    defaultVariant?: keyof Props;
    base?: CSSProps<Base, SystemProps<P>>;
    variants: CSSPropMap<Props, SystemProps<P>>;
  }) {
    type NextVariants = Record<PropKey, typeof options>;
    const prop = options.prop || 'variant';

    return new AnimusWithVariants(
      this.props,
      this.parser,
      this.base,
      merge(this.variants, { [prop]: options }) as NextVariants
    );
  }
}

export class Animus<
  C extends PropConfig,
  P extends Parser<TransformerMap<C['props']>>
> extends AnimusWithBase<C, P, {}> {
  constructor(config: C) {
    super(config, create(config.props) as P, {});
  }
  styles<Props extends AbstractProps>(config: CSSProps<Props, SystemProps<P>>) {
    return new AnimusWithBase(this.props, this.parser, config);
  }
}
