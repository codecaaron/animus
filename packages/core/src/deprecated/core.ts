import { createCss } from '../internal/createCss';
import { createVariant } from '../internal/createVariant';
import { create } from '../internal/create';
import { compose } from '../internal/compose';

export const animusProps = {
  compose,
  create,
  createCss,
  createVariant,
};
