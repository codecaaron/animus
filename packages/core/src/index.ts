export * from './types/props';
export * from './types/theme';
export * from './types/utils';
export * from './types/shared';

export * from './scales/createScale';
export * from './AnimusConfig';
export * from './createAnimus';
export * from './transforms';
export * from './legacy/core';

export { config } from './config';
import { config } from './config';

export const animus = config.build();
animus
  .styles({
    size: '2px',
  })
  .variant({
    prop: 'cool',
    variants: {
      many: {
        fontSize: 14,
      },
    },
  });
