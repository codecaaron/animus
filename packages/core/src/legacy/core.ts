import { createCss } from './createCss';
import { createVariant } from './createVariant';
import { create } from './create';
import { compose } from './compose';
import { createStates } from './createStates';

export const animusProps = {
  compose,
  create,
  createCss,
  createVariant,
  createStates,
};
