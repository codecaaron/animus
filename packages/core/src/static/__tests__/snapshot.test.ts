import { extractAndGenerateCSS } from '../index';

describe('Animus Static Extraction Integration', () => {
  it('generates CSS for complex Button and Card components', () => {
    const code = `
import { animus } from '@animus-ui/core';

const Button = animus
  .styles({
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 12,
    fontSize: 16,
    fontWeight: 500,
    borderRadius: 4,
    cursor: 'pointer',
    transition: 'all 0.2s ease',
    backgroundColor: '#007bff',
    color: 'white',
    border: 'none',
    '&:hover': {
      backgroundColor: '#0056b3',
      transform: 'translateY(-1px)',
      boxShadow: '0 4px 8px rgba(0, 0, 0, 0.1)'
    },
    '&:active': {
      transform: 'translateY(0)',
      boxShadow: '0 2px 4px rgba(0, 0, 0, 0.1)'
    }
  })
  .variant({
    prop: 'size',
    variants: {
      small: { 
        padding: 8,
        fontSize: 14 
      },
      large: { 
        padding: 16,
        fontSize: 18 
      }
    }
  })
  .states({
    disabled: { 
      opacity: 0.6, 
      cursor: 'not-allowed',
      '&:hover': {
        backgroundColor: '#007bff',
        transform: 'none'
      }
    }
  })
  .asElement('button');

const Card = animus
  .styles({
    backgroundColor: 'white',
    borderRadius: 8,
    padding: 20,
    boxShadow: '0 2px 4px rgba(0, 0, 0, 0.1)',
    transition: 'box-shadow 0.2s ease',
    '&:hover': {
      boxShadow: '0 4px 12px rgba(0, 0, 0, 0.15)'
    }
  })
  .asElement('div');
`;

    const result = extractAndGenerateCSS(code);
    expect(result).toMatchSnapshot('button-card-integration');
  });

  it('generates CSS for component with groups and props', () => {
    const code = `
const Box = animus
  .styles({
    display: 'flex',
    flexDirection: 'column',
    gap: 16,
  })
  .groups({
    space: true,
    color: true,
  })
  .props({
    cols: {
      property: 'gridTemplateColumns',
      scale: 'grid',
    }
  })
  .asElement('div');
`;

    const result = extractAndGenerateCSS(code);
    expect(result).toMatchSnapshot('groups-props-integration');
  });

  it('generates CSS for responsive component', () => {
    const code = `
const ResponsiveBox = animus
  .styles({
    padding: { _: 10, sm: 20, lg: 40 },
    margin: [5, 10, 15],
    fontSize: 16,
    display: 'flex',
    gap: { _: 8, md: 16 },
  })
  .asElement('div');
`;

    const result = extractAndGenerateCSS(code);
    expect(result).toMatchSnapshot('responsive-integration');
  });

  it('generates CSS for complex nested selectors', () => {
    const code = `
const NavLink = animus
  .styles({
    display: 'block',
    padding: '8px 16px',
    color: 'gray',
    textDecoration: 'none',
    transition: 'all 0.2s',
    '&:hover': {
      color: 'blue',
      backgroundColor: 'rgba(0, 0, 255, 0.1)',
    },
    '&:active': {
      backgroundColor: 'rgba(0, 0, 255, 0.2)',
    },
    '&:focus': {
      outline: '2px solid blue',
      outlineOffset: 2,
    },
    '&.active': {
      color: 'darkblue',
      fontWeight: 600,
      '&:hover': {
        backgroundColor: 'rgba(0, 0, 255, 0.15)',
      }
    }
  })
  .asElement('a');
`;

    const result = extractAndGenerateCSS(code);
    expect(result).toMatchSnapshot('nested-selectors-integration');
  });
});
