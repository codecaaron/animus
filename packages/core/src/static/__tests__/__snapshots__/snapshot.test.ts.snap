// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Animus Static Extraction Integration generates CSS for complex Button and Card components: button-card-integration 1`] = `
{
  "allCSS": "/* Base Styles */
.animus-Button-b6n {
  padding: 12px;
  transition: all 0.2s ease;
  border: none;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  font-size: 16px;
  font-weight: 500;
  border-radius: 4px;
  cursor: pointer;
  background-color: #007bff;
  color: white;
}
.animus-Button-b6n:hover {
  background-color: #0056b3;
  transform: translateY(-1px);
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
}
.animus-Button-b6n:active {
  transform: translateY(0);
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

/* Variant: size="small" */
.animus-Button-b6n-size-small {
  padding: 8px;
  font-size: 14px;
}

/* Variant: size="large" */
.animus-Button-b6n-size-large {
  padding: 16px;
  font-size: 18px;
}

/* State: disabled */
.animus-Button-b6n-state-disabled {
  opacity: 0.6;
  cursor: not-allowed;
}
.animus-Button-b6n-state-disabled:hover {
  background-color: #007bff;
  transform: none;
}

/* Base Styles */
.animus-Card-c4d {
  padding: 20px;
  transition: box-shadow 0.2s ease;
  background-color: white;
  border-radius: 8px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}
.animus-Card-c4d:hover {
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
}",
  "components": [
    {
      "className": "animus-Button-b6n",
      "css": "/* Base Styles */
.animus-Button-b6n {
  padding: 12px;
  transition: all 0.2s ease;
  border: none;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  font-size: 16px;
  font-weight: 500;
  border-radius: 4px;
  cursor: pointer;
  background-color: #007bff;
  color: white;
}
.animus-Button-b6n:hover {
  background-color: #0056b3;
  transform: translateY(-1px);
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
}
.animus-Button-b6n:active {
  transform: translateY(0);
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

/* Variant: size="small" */
.animus-Button-b6n-size-small {
  padding: 8px;
  font-size: 14px;
}

/* Variant: size="large" */
.animus-Button-b6n-size-large {
  padding: 16px;
  font-size: 18px;
}

/* State: disabled */
.animus-Button-b6n-state-disabled {
  opacity: 0.6;
  cursor: not-allowed;
}
.animus-Button-b6n-state-disabled:hover {
  background-color: #007bff;
  transform: none;
}",
      "name": "Button",
    },
    {
      "className": "animus-Card-c4d",
      "css": "/* Base Styles */
.animus-Card-c4d {
  padding: 20px;
  transition: box-shadow 0.2s ease;
  background-color: white;
  border-radius: 8px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}
.animus-Card-c4d:hover {
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
}",
      "name": "Card",
    },
  ],
}
`;

exports[`Animus Static Extraction Integration generates CSS for complex nested selectors: nested-selectors-integration 1`] = `
{
  "allCSS": "/* Base Styles */
.animus-NavLink-n7k {
  padding: 8px 16px;
  transition: all 0.2s;
  display: block;
  color: gray;
  text-decoration: none;
}
.animus-NavLink-n7k:hover {
  color: blue;
  background-color: rgba(0, 0, 255, 0.1);
}
.animus-NavLink-n7k:active {
  background-color: rgba(0, 0, 255, 0.2);
}
.animus-NavLink-n7k:focus {
  outline: 2px solid blue;
  outline-offset: 2;
}
.animus-NavLink-n7k.active {
  color: darkblue;
  font-weight: 600;
}
.animus-NavLink-n7k.active:hover {
  background-color: rgba(0, 0, 255, 0.15);
}",
  "components": [
    {
      "className": "animus-NavLink-n7k",
      "css": "/* Base Styles */
.animus-NavLink-n7k {
  padding: 8px 16px;
  transition: all 0.2s;
  display: block;
  color: gray;
  text-decoration: none;
}
.animus-NavLink-n7k:hover {
  color: blue;
  background-color: rgba(0, 0, 255, 0.1);
}
.animus-NavLink-n7k:active {
  background-color: rgba(0, 0, 255, 0.2);
}
.animus-NavLink-n7k:focus {
  outline: 2px solid blue;
  outline-offset: 2;
}
.animus-NavLink-n7k.active {
  color: darkblue;
  font-weight: 600;
}
.animus-NavLink-n7k.active:hover {
  background-color: rgba(0, 0, 255, 0.15);
}",
      "name": "NavLink",
    },
  ],
}
`;

exports[`Animus Static Extraction Integration generates CSS for component with groups and props: groups-props-integration 1`] = `
{
  "allCSS": "/* Base Styles */
.animus-Box-b3x {
  gap: 16px;
  display: flex;
  flex-direction: column;
}


/* Atomic Utilities from Custom Props */",
  "components": [
    {
      "className": "animus-Box-b3x",
      "css": "/* Base Styles */
.animus-Box-b3x {
  gap: 16px;
  display: flex;
  flex-direction: column;
}


/* Atomic Utilities from Custom Props */",
      "name": "Box",
    },
  ],
}
`;

exports[`Animus Static Extraction Integration generates CSS for responsive component: responsive-integration 1`] = `
{
  "allCSS": "/* Base Styles */
.animus-ResponsiveBox-r13x {
  padding: 10px;
  margin: 5px;
  gap: 8px;
  font-size: 16px;
  display: flex;
}
@media screen and (min-width: 768px) {
  .animus-ResponsiveBox-r13x {
    padding: 20px;
    margin: 15px;
  }
}
@media screen and (min-width: 1200px) {
  .animus-ResponsiveBox-r13x {
    padding: 40px;
  }
}
@media screen and (min-width: 480px) {
  .animus-ResponsiveBox-r13x {
    margin: 10px;
  }
}
@media screen and (min-width: 1024px) {
  .animus-ResponsiveBox-r13x {
    gap: 16px;
  }
}",
  "components": [
    {
      "className": "animus-ResponsiveBox-r13x",
      "css": "/* Base Styles */
.animus-ResponsiveBox-r13x {
  padding: 10px;
  margin: 5px;
  gap: 8px;
  font-size: 16px;
  display: flex;
}
@media screen and (min-width: 768px) {
  .animus-ResponsiveBox-r13x {
    padding: 20px;
    margin: 15px;
  }
}
@media screen and (min-width: 1200px) {
  .animus-ResponsiveBox-r13x {
    padding: 40px;
  }
}
@media screen and (min-width: 480px) {
  .animus-ResponsiveBox-r13x {
    margin: 10px;
  }
}
@media screen and (min-width: 1024px) {
  .animus-ResponsiveBox-r13x {
    gap: 16px;
  }
}",
      "name": "ResponsiveBox",
    },
  ],
}
`;
